stdlib/_typeshed/__init__.pyi

```diff
@@ -328,9 +328,9 @@ class structseq(Generic[_T_co]):
     # The second parameter will accept a dict of any kind without raising an exception,
     # but only has any meaning if you supply it a dict where the keys are strings.
     # https://github.com/python/typeshed/pull/6560#discussion_r767149830
-    def __new__(cls: type[Self], sequence: Iterable[_T_co], dict: dict[str, Any] = ...) -> Self: ...
+    def __new__(cls: type[Self], sequence: Iterable[_T_co], dict: dict[str, Any] = ...) -> Self: ...  # noqa: PYI019
     if sys.version_info >= (3, 13):
-        def __replace__(self: Self, **kwargs: Any) -> Self: ...
+        def __replace__(self: Self, **kwargs: Any) -> Self: ...  # noqa: PYI019
```

Comment:
We could probably do a similar thing to typing.pyi here and use the fully qualified `typing_extensions.Self` to avoid having to add these noqa comments?

Reply:
I'll try it. Maybe `Self = TypeVar("Self")` at line 40 isn't needed anymore.

Reply:
> Maybe `Self = TypeVar("Self")` at line 40 isn't needed anymore.

I think it is, because we use it in other modules for metaclasses (`typing.Self` is not allowed in metaclass methods)

Reply:
From a quick search it's still used in 2 openpyxl modules for metaclasses.